Fixing and enabling tests that check fake_quant matches quant+dequant

HDCharles opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59095 Fixing and enabling tests that check fake_quant matches quant+dequant**

Summary: These tests were disabled, I'm unsure as to why. I've
re-enabled them and remade them to expand testing to different devices
and dtypes

Test Plan: python test/test_quantization.py TestFakeQuantizeOps.test_numerical_consistency

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018745](https://our.internmc.facebook.com/intern/diff/D29018745)

HDCharles committed:
Fixing and enabling tests that check fake_quant matches quant+dequant

Summary: These tests were disabled, I'm unsure as to why. I've
re-enabled them and remade them to expand testing to different devices
and dtypes

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "Fixing and enabling tests that check fake_quant matches quant+dequant"


Summary: These tests were disabled, I'm unsure as to why. I've
re-enabled them and remade them to expand testing to different devices
and dtypes

Test Plan: python test/test_quantization.py TestFakeQuantizeOps.test_numerical_consistency

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "Fixing and enabling tests that check fake_quant matches quant+dequant"


Summary: These tests were disabled, I'm unsure as to why. I've
re-enabled them and remade them to expand testing to different devices
and dtypes

Test Plan: python test/test_quantization.py TestFakeQuantizeOps.test_numerical_consistency

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on test/quantization/core/test_workflow_ops.py:
```diff
@@ -840,26 +820,32 @@ def test_learnable_backward_per_channel_cuda(self, X):
-        Y_prime = torch.fake_quantize_per_channel_affine(
-            X, scale, zero_point, axis, quant_min, quant_max)
-        np.testing.assert_allclose(Y, Y_prime.cpu(), rtol=tolerance, atol=tolerance)
+        torch.random.manual_seed(NP_RANDOM_SEED)
```
nit: this should already be set by the testing framework, we can probably remove it

HDCharles replied:
it isn't. I just tested it, get different outputs with/without it

vkuzo commented on test/quantization/core/test_workflow_ops.py:
```diff
@@ -840,26 +820,32 @@ def test_learnable_backward_per_channel_cuda(self, X):
+                Y_prime = torch.fake_quantize_per_tensor_affine(X, scale, zero, quant_min, quant_max)
+                self.assertEqual(Y, Y_prime, "Difference found between dequant+quant_per_tensor and fake_quantize_per_tensor")
+
+                Y = torch.dequantize(torch.quantize_per_channel(X.to('cpu').to(torch.float), scales.to('cpu'), zeros.to('cpu'), axis, torch_type)).to(device).to(float_type)
```
optional nit: would be nice to have per-tensor and per-channel in individual test cases instead of together, to keep tests single purposed

HDCharles committed:
Update on "Fixing and enabling tests that check fake_quant matches quant+dequant"


Summary: These tests were disabled, I'm unsure as to why. I've
re-enabled them and remade them to expand testing to different devices
and dtypes

Test Plan: python test/test_quantization.py TestFakeQuantizeOps.test_numerical_consistency

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "Fixing and enabling tests that check fake_quant matches quant+dequant"


Summary: These tests were disabled, I'm unsure as to why. I've
re-enabled them and remade them to expand testing to different devices
and dtypes

Test Plan: python test/test_quantization.py TestFakeQuantizeOps.test_numerical_consistency

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018745).
